packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,107 @@
+| content       | `string`                                                                 | `undefined` | Content of the tooltip                                                    | ✅       |
+| placement     | `top, top-start, top-end, left, right, bottom, bottom-start, bottom-end` | `top`       | Placement of tooltip                                                      | ✅       |
+| children      | `React.ReactNode`                                                        | `undefined` | Trigger component for tooltip, Accepts any interactive element            | ✅       |
+| isOpen        | `boolean`                                                                | `undefined` | Controls the tooltip state                                                |          |
+| defaultIsOpen | `boolean`                                                                | `undefined` | If true, the tooltip will be visible initially                            |          |
+| onDismiss     | `Callback`                                                               | `undefined` | Called when tooltip is closed, either by mouseout or users pressing `Esc` |          |
+
+### Examples:
+
+#### Usage
+
+```js
+// basic, uncontrolled
+<Tooltip content="Amount reversed to customer bank account" placement="top">
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>;
+
+// controlled
+const [isOpen, setIsOpen] = React.useState(false);
+
+<Tooltip
+  onDismiss={() => setIsOpen(false)}
+  isOpen={isOpen}
+  content="Amount reversed to customer bank account"
+>
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>;
+
+// default open
+<Tooltip defaultIsOpen={true} content="Amount reversed to customer bank account">
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>;
+```
+
+## Library
+
+We will be using [FloatingUI](https://floating-ui.com/) to position the tooltip & handle the basic tooltip logic. Floating UI supports both Web & ReactNative.
+
+## Implementation detail nuances:
+
+To make FloatingUI work with Blade components seamlessly there are few things we need to modify:
+
+1. Expose interaction props in all the trigger components
+- Web:
+  - onBlur
+  - onFocus
+  - onMouseLeave
+  - onMouseMove
+  - onPointerDown
+  - onPointerEnter
+- Native:
+  - onTouchStart
+  - onTouchEnd
+
+2. Expose the actual DOM node from ref instead of only exposing [certain methods via our useBladeInnerRef](https://github.com/razorpay/blade/blob/69a1bcef2f09ceaf6f910eaaca3076055fb059a2/packages/blade/src/hooks/useBladeInnerRef.web.ts#L26-L27) hook which we used to prevent component styling misuses. We need to expose the actual DOM node because FloatingUI internally does this [isElement() check](https://github.com/floating-ui/floating-ui/blob/b8990250568043b876e1c8fe42358fe337847ede/packages/react/src/hooks/useFloating.ts#L59) on the tooltip trigger element.
```

Comment:
I feel this might still be okay. Restricting methods on `ref` doesn't restrict users from using `document.` APIs to access the DOM node directly anyway.